react/features/base/react/components/native/SideBar.js

```diff
@@ -0,0 +1,204 @@
+/* @flow */
+
+import React, { Component } from 'react';
+import {
+    Animated,
+    Dimensions,
+    TouchableWithoutFeedback,
+    View
+} from 'react-native';
+
+import styles, { SIDEBAR_WIDTH } from './styles';
+
+
+/**
+ * The type of the React {@code Component} props of {@link AnimatedSideBar}
+ */
+type Props = {
+
+    /**
+     * The local participant's avatar
+     */
+    _avatar: string,
+
+    /**
+     * The children of the Component
+     */
+    children: Object,
+
+    /**
+     * Callback to notify the containing Component that the sidebar is
+     * closing.
+     */
+    onHide: Function,
+
+    /**
+     * Sets the menu displayed or hidden.
+     */
+    show: boolean
+}
+
+type State = {
+
+    /**
+     * Indicates whether the side bar is visible or not.
+     */
+    showSideBar: boolean,
+
+    /**
+     * Indicates whether the side overlay should be rendered or not.
+     */
+    showOverlay: boolean,
+
+    /**
+     * The native animation object.
+     */
+    sliderAnimation: Object
+}
+
+/**
+ * A generic animated side bar to be used for left side menus
+ */
+export default class SideBar extends Component<Props, State> {
+    _ismounted: boolean;
+
+    /**
+     * Component's contructor.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            showSideBar: false,
+            showOverlay: false,
+            sliderAnimation: new Animated.Value(-SIDEBAR_WIDTH)
+        };
+
+        this._setShow = this._setShow.bind(this);
+
```

Comment:
why the newline?

Reply:
I like to separate logically disjoined blocks, like this one is a behaviour call, and the next 3 lines are function bindings. For me they don't really belong together